plugin/storage/es/spanstore/writer.go

```diff
@@ -107,16 +107,19 @@ func (s *SpanWriter) WriteSpan(span *model.Span) error {
 
 func spanIndexName(span *model.Span) string {
 	spanDate := span.StartTime.Format("2006-01-02")
-	return "jaeger-" + spanDate
+	return indexPrefix + spanDate
 }
 
 func (s *SpanWriter) createIndex(indexName string, jsonSpan *jModel.Span) error {
 	if !keyInCache(indexName, s.indexCache) {
 		start := time.Now()
-		_, err := s.client.CreateIndex(indexName).Body(spanMapping).Do(s.ctx)
-		s.writerMetrics.indexCreate.Emit(err, time.Since(start))
-		if err != nil {
-			return s.logError(jsonSpan, err, "Failed to create index", s.logger)
+		exists, _ := s.client.IndexExists(indexName).Do(s.ctx) // don't need to check the error because the exists variable will be false anyway if there is an error
```

Comment:
add a comment around here detailing that we might drop a couple of spans every time a new index is created.